modules/subscriptions.php

```diff
@@ -161,6 +163,16 @@ function maybe_send_subscription_email( $new_status, $old_status, $post ) {
 			update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', 1 );
 		}
 
+		if ( ! $this->should_email_post_to_subscribers( $post ) ) {
+			update_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', 1 );
+		}
+	}
+
+	public function should_email_post_to_subscribers( $post ) {
+		if ( get_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs' ) ) {
```

Comment:
this should be `get_post_meta( $post->ID, '_jetpack_dont_email_post_to_subs', true )`